### PR TITLE
Refactor PostgreSQLExecutor to support Windows compatibility for process management

### PR DESCRIPTION
**Pull request by tboy1337**
Refactors `PostgreSQLExecutor` to improve Windows compatibility and security for PostgreSQL process management. This change addresses platform-specific issues with process termination and command formatting that could cause failures on Windows systems.

## Changes
- **Unified command format**: Converted `_get_base_command` method to `BASE_PROC_START_COMMAND` class attribute with cross-platform compatible syntax (removed single quotes around PostgreSQL config values that caused issues on Windows)
- **Graceful Windows termination**: Added `_windows_terminate_process` method that attempts graceful shutdown before force-killing stuck processes
- **Improved security**: Updated `stop()` method to use list arguments instead of `shell=True` to prevent shell injection vulnerabilities
- **Platform-specific logic**: Added intelligent platform detection with fallback handling for `os.killpg` AttributeError on Windows
- **Comprehensive test coverage**: Added 186 lines of Windows compatibility tests covering various termination scenarios and command formatting

## Test plan
- ✅ New test suite in `tests/test_windows_compatibility.py` with 7 test cases
- ✅ Tests verify unified command format without single quotes
- ✅ Tests verify graceful termination and force-kill timeout handling
- ✅ Tests verify platform-specific stop behavior (Windows vs Unix)
- ✅ Tests verify fallback logic for `killpg` AttributeError
- ✅ Tests verify command formatting with Windows paths

## Summary by CodeRabbit

* **Improvements**
  * Cross‑platform PostgreSQL start/stop: safer subprocess use, synchronous Windows start, explicit shutdown escalation, merged environment handling, timeout/error surfacing, and robust quoting/escaping for complex paths/options.

* **Tests**
  * Extensive Windows and cross‑platform tests for command generation, quoting/escaping edge cases, lifecycle behaviour, failure and fallback scenarios, and integration start/stop checks.

* **CI**
  * Added Windows PostgreSQL CI workflow, extended Windows job matrix, and dynamic detection of the pg_ctl executable.

* **Documentation**
  * Added a news fragment announcing Windows compatibility updates.

### CODE REVIEW

**Review by fizyk — APPROVED** [review]

**Review by fizyk — COMMENTED** [review]
@tboy1337 are we able to add ci windows-based tests?

File: .github/workflows/single-postgres.yml [hunk 1]
```diff
@@ -43,29 +43,63 @@ jobs:
...
+            $pgPath = (Get-Command pg_ctl -ErrorAction SilentlyContinue).Source
+            if ($pgPath) {
+              echo "POSTGRESQL_EXEC=$pgPath" >> $env:GITHUB_ENV
+            }
+          }
+      - name: Set PostgreSQL path for Unix/macOS
+        if: runner.os != 'Windows'
+        run: |
+          # Try to find pg_ctl dynamically for cross-platform compatibility
+          if command -v pg_ctl >/dev/null 2>&1; then
```
Comment (fizyk) [inline .github/workflows/single-postgres.yml]: pytest-postgresql can do that detection as well in case of wrong path :)
Maybe that's why I haven't bothered with specifying separate path for macos

**Review by fizyk — APPROVED** [review]

**Comment by tboy1337** [comment]
@coderabbitai resume

**Comment by tboy1337** [comment]
ok, so this solution seems to be the best, also i added -e method to the ci, is this ok?

**Comment by tboy1337** [comment]
@coderabbitai resume

File: tests/test_executor.py [hunk 2]
```diff
@@ -119,6 +121,114 @@ def test_executor_init_bad_tmp_path(
...
+
+    executor = PostgreSQLExecutor(
+        executable=pg_exe,
+        host=config.host,
+        port=port,
+        datadir=str(datadir),
+        unixsocketdir=str(socket_dir),
+        logfile=str(logfile_path),
+        startparams=config.startparams,
+        password="test pass",
```
[the remark this answers is not in the document]
Comment (tboy1337) [reply tests/test_executor.py]: Ruff's `select` in `pyproject.toml` is `["E", "F", "I", "D"]`. The `S` group (flake8-bandit) is not included, so S106 never fires. No `# noqa` comments or per-file ignores are needed.

File: tests/test_windows_compatibility.py
Comment (fizyk) [inline tests/test_windows_compatibility.py]: Those tests should only fire for windows, right? Use https://docs.pytest.org/en/stable/how-to/skipping.html#id1 to skip... maybe whole module if tests are not being run on windows

**Comment by fizyk** [comment]
@tboy1337 I've merged main into your branch to have the tests run

**Comment by tboy1337** [comment]
@fizyk should be ok now.

**Comment by tboy1337** [comment]
@fizyk ok maybe now 😁

File: .github/workflows/oldest-postgres.yml [hunk 3]
```diff
@@ -52,7 +52,7 @@ jobs:
         run: |
           sudo locale-gen de_DE.UTF-8
       - name: install libpq
-        if: ${{ contains(inputs.python-versions, 'pypy') }}
+        if: ${{ contains(matrix.python-version, 'pypy') && runner.os == 'Linux' }}
```
Comment (fizyk) [inline .github/workflows/oldest-postgres.yml:55]: you can drop the runner.os condition, we're not going to run the oldest on Windows, at least not yet.

File: .github/workflows/single-postgres.yml [hunk 4]
```diff
@@ -43,29 +43,69 @@ jobs:
       - uses: ankane/setup-postgres@v1
         with:
           postgres-version: ${{ inputs.postgresql }}
+      - name: Detect PostgreSQL path on Windows
```
Comment (fizyk) [inline .github/workflows/single-postgres.yml]: hmmm... maybe we should just branch it off into single-postgres-windows.yml worflow instead. Could be cleaner to maintain later

**Comment by tboy1337** [comment]
@coderabbitai resume

**Comment by tboy1337** [comment]
@coderabbitai resume

**Comment by tboy1337** [comment]
@coderabbitai resume

File: .github/workflows/single-postgres-windows.yml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      PYTHON: ${{ matrix.python-version }}
+      POSTGRES: ${{ inputs.postgresql }}
+    steps:
+      - uses: actions/checkout@v6
+      - name: Set up Pipenv on python ${{ matrix.python-version }}
+        uses: fizyk/actions-reuse/.github/actions/pipenv-setup@v4.4.4
+        with:
+          python-version: ${{ matrix.python-version }}
+          allow-prereleases: true
+      - uses: ankane/setup-postgres@v1
```
[the remark this answers is not in the document]
Comment (fizyk) [reply .github/workflows/single-postgres-windows.yml:38]: @coderabbitai if dependabot handles it, defer that to separate issue

File: tests/test_windows_compatibility.py [hunk 6]
```diff
@@ -0,0 +1,894 @@
...
+        with (
+            patch("pytest_postgresql.executor.subprocess.check_output") as mock_subprocess,
+            patch("pytest_postgresql.executor.platform.system", return_value="Windows"),
+            patch.object(executor, "_windows_terminate_process") as mock_terminate,
+        ):
+            result = executor.stop()
+
+            # Should call pg_ctl stop and Windows terminate
+            mock_subprocess.assert_called_once_with(
+                ["/path/to/pg_ctl", "stop", "-D", "/tmp/data", "-m", "f"],
```
Comment (fizyk) [inline tests/test_windows_compatibility.py:594]: If I'm not mistaken, the timeout is missing, and we'll be all set.

File: .github/workflows/single-postgres-windows.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (fizyk) [reply .github/workflows/single-postgres-windows.yml:38]: yes, that's what I asked. Not sure how that would help, with my resources being a dependabot and auto-merge, with no real time for manual review od those updates. Well at least setup-posgres updates would be more visible.

[the remark this answers is not in the document]
Comment (tboy1337) [reply .github/workflows/single-postgres-windows.yml:38]: https://github.com/dbfixtures/pytest-postgresql/issues/1331

**Review by fizyk — APPROVED** [review]

**Comment by fizyk** [comment]
Thanks @tboy1337 !

**Comment by cianjm** [comment]
Thanks so much for this! It's a huge help to be able to run this on Windows directly
